executor/index_merge_reader.go

```diff
@@ -1249,6 +1253,73 @@ func (w *indexMergeProcessWorker) fetchLoopUnion(ctx context.Context, fetchCh <-
 	}
 }
 
+// intersectionCollectWorker is used to dispatch index-merge-table-task to original workCh and resultCh.
+// a kind of interceptor to control the pushed down limit restriction. (should be no performance impact)
+type intersectionCollectWorker struct {
+	pushedLimit *plannercore.PushedDownLimit
+	collectCh   chan *indexMergeTableTask
+	limitDone   chan struct{}
+}
+
+func (w *intersectionCollectWorker) doIntersectionLimitAndDispatch(ctx context.Context, workCh chan<- *indexMergeTableTask,
+	resultCh chan<- *indexMergeTableTask, finished <-chan struct{}) {
+	var (
+		ok   bool
+		task *indexMergeTableTask
+	)
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-finished:
+			return
+		case task, ok = <-w.collectCh:
+			if !ok {
+				return
+			}
+			// receive a new intersection task here, adding limit restriction logic
+			if w.pushedLimit != nil {
```

Comment:
`partialIndexWorker.pushedLimit` also set and will be used in `partialIndexWorker.extractTaskHandles()`, we should handle it.

Reply:
make sense, addressed, for intersection case, the pushdown limit can only be conducted after all partial path is done

Reply:
Better add a test case for this case.